editor/inserter/index.js

```diff
@@ -1,18 +1,31 @@
+/**
+ * External dependencies
+ */
+import { groupBy } from 'lodash';
+
 const Inserter = () => {
 	const blocks = wp.blocks.getBlocks();
+	const blocksByCategory = groupBy( blocks, 'category' );
+	const categories = wp.blocks.getCategories();
 
 	return (
 		<div className="inserter">
 			<div className="inserter__arrow" />
 			<div className="inserter__content">
-				<div className="inserter__Category-blocks">
-					{ blocks.map( ( { slug, title, icon } ) => (
-						<div key={ slug } className="inserter__block">
-							<span className={ 'dashicons dashicons-' + icon } />
-							{ title }
+				{ categories
+					.filter( ( category ) => !! blocksByCategory[ category.id ] )
```

Comment:
Ever a time there's multiple iterators like here, I'm inclined to nudge toward `Array#reduce` instead.